src/block/Netherrack.php

```diff
@@ -23,9 +23,41 @@
 
 namespace pocketmine\block;
 
+use pocketmine\item\Fertilizer;
+use pocketmine\item\Item;
+use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
+
 class Netherrack extends Opaque{
 
 	public function burnsForever() : bool{
 		return true;
 	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if($this->getSide(Facing::UP)->getTypeId() !== BlockTypeIds::AIR){
+			return false;
+		}
+
+		if($item instanceof Fertilizer && $this->transformToNylium()){
+			$item->pop();
+
+			return true;
+		}
+
+		return false;
+	}
+
+	private function transformToNylium() : bool{
+		foreach($this->getHorizontalSides() as $side){
+			if($side instanceof Nylium){
+				$this->position->getWorld()->setBlock($this->position, $side);
```

Comment:
Why did you use `$side` here, couldn't be better to use a fresh instance ?

Reply:
Don´t know what you mean

Reply:
how should I do it instead? anyways the block state never changes

Reply:
`	private function transformToNylium() : bool{
		foreach($this->getHorizontalSides() as $side){
			if($side->hasTypeTag(BlockTypeTags::NYLIUM)){
				$block = null;
				if($side->getTypeId() === BlockTypeIds::CRIMSON_NYLIUM){
					$block = VanillaBlocks::CRIMSON_NYLIUM();
				}elseif($side->getTypeId() === BlockTypeIds::WARPED_NYLIUM){
					$block = VanillaBlocks::WARPED_NYLIUM();
				}
				
				$this->position->getWorld()->setBlock($this->position, $block);
				
				return true;
			}
		}

		return false;
	}`
	
	Do you like this code more?

Reply:
My fault, the original code looks ok